addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -2822,6 +2822,165 @@ options.registry.ReplaceImage = options.Class.extend({
+        }
+        if (ids.length && values.length) {
+            this.$target[0].dataset[`${params.attributeName}Id`] = ids.join(',');
+            this.$target[0].dataset[params.attributeName] = values.join(',');
+        } else {
+            delete this.$target[0].dataset[params.attributeName];
+            delete this.$target[0].dataset[`${params.attributeName}Id`];
+        }
+    },
+
+    /**
+     * extends data attribute to open the toggler when conditionnal is selected
+     * @override
+     */
+    selectDataAttribute(previewMode, widgetValue, params) {
+        this._super(...arguments);
+        if (params.attributeName === 'visibility' && widgetValue === 'conditional') {
+            const collapseEl = this.$el.children('we-collapse')[0];
+            this._toggleCollapseEl(collapseEl);
+        }
+    },
+    /**
+     * @override
+     */
+    async onTargetHide() {
+        this.$target[0].previousStyle = this.$target[0].style;
+        this.$target[0].style.display = 'none';
+    },
+    /**
+     * @override
+     */
+    async onTargetShow() {
+        this.$target[0].style = this.$target[0].previousStyle;
+    },
+
+
+    /**
+     * Reades the attributes of the traget and creates the selectors
+     * Store them in data-attributes to then be reapplied by content/inject_dom.js
+     * @override
+    */
+    async cleanForSave() {
+        this.$target[0].style = this.$target[0].previousStyle;
+        delete this.$target[0].dataset.invisible;
+        if (this.$target[0].dataset.visibility !== 'conditional') {
+            delete this.$target[0].dataset.visibilityId;
+            delete this.$target[0].dataset.visibilityRule;
+            return this._super(...arguments);
+        }
+        let visibilityId = '';
+        const onlyAttributes = [];
+        const hideAttributes = [];
+        for (const attr of this.$target[0].attributes) {
+            if (attr.name.startsWith('data-visibility-value-') && !attr.name.endsWith('-id')) {
+                const shortName = attr.name.replace('data-visibility-value-', '');
+                let dataName, values;
+                if (shortName === 'lang') {
+                    dataName = 'lang';
+                    values = attr.value.replace(/_/g, '-').split(',');
+                } else {
+                    dataName = `data-${shortName}`;
+                    values = attr.value.split(',');
+                }
+                const hideFor = this.$target[0].attributes[`data-visibility-rule-${shortName}`] && this.$target[0].attributes[`data-visibility-rule-${shortName}`].value === 'hide';
+                if (hideFor) {
+                    hideAttributes.push({ name: dataName, values: values });
+                } else {
+                    onlyAttributes.push({ name: dataName, values: values });
+                }
+                visibilityId += shortName + this.$target[0].attributes[`data-visibility-value-${shortName}-id`].value + (hideFor ? 'h' : '');
```

Comment:
I am not sure to understand this whole code, can you explain the different attributes you are reading? There seems to be data-visibility-value-XXX and data-visibility-value-XXX-id... and I don't see where and why is the difference.